cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,188 @@
+
+    def test_movie_detail(self):
+        response = self.client.get(movie_detail(self.movie_1.id))
+        serializer = MovieDetailSerializer(self.movie_1, many=False)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_movie_filter_by_title(self):
+        response = self.client.get(MOVIE_URL, {"title": f"{self.movie_2.title}"})
+        serializer_1 = MovieListSerializer(self.movie_1, many=False)
+        serializer_2 = MovieListSerializer(self.movie_2, many=False)
+
+        self.assertIn(serializer_2.data, response.data)
+
+    def test_movie_filter_by_genres(self):
+        movie_3 = sample_movie()
+        genre_3 = Genre.objects.create(name="genre_3")
+        movie_3.genres.set([genre_3])
+        movie_3.save()
+
+        serializer_3 = MovieListSerializer(movie_3, many=False)
+
+        serializer_1 = MovieListSerializer(self.movie_1, many=False)
+        serializer_2 = MovieListSerializer(self.movie_2, many=False)
+
+        response = self.client.get(
+            MOVIE_URL, {"genres": f"{self.genre_2.id}, {self.genre.id}"}
+        )
+
+        self.assertIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)
+        self.assertNotIn(serializer_3.data, response.data)
+
+    def test_movie_filter_by_actors(self):
+        movie_3 = sample_movie()
+        actor_3 = Actor.objects.create(first_name="<NAME>", last_name="<NAME>")
+        movie_3.actors.set([actor_3])
+        movie_3.save()
+
+        serializer_3 = MovieListSerializer(movie_3, many=False)
+
+        serializer_1 = MovieListSerializer(self.movie_1, many=False)
+        serializer_2 = MovieListSerializer(self.movie_2, many=False)
+
+        response = self.client.get(
+            MOVIE_URL, {"actors": f"{self.actor.id}, {self.actor_2.id}"}
+        )
+
+        self.assertIn(serializer_1.data, response.data)
+        self.assertIn(serializer_2.data, response.data)
+        self.assertNotIn(serializer_3.data, response.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "Test title",
+            "description": "Test description",
+            "duration": 123,
+            "genres": self.genre_2,
+            "actors": self.actor_2,
+        }
+
+        response = self.client.post(MOVIE_URL, payload)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.com",
+            password="<PASSWORD>123",
+            is_staff=True,
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movie_creation_accessed(self):
+        genre = Genre.objects.create(name="Test genre")
+        actor = Actor.objects.create(
+            first_name="Sample first_name",
+            last_name="Sample last_name",
+        )
```

Comment:
In case you create similar objects in each test, move this operation to `setUp` for deduplication